MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -880,20 +880,35 @@ - (void)computeDrift {
     float playheadTimeElapsed = ([self getCurrentPlayheadTimeMs] - _lastPlayheadTimeMs) / 1000;
     float wallTimeElapsed = CFAbsoluteTimeGetCurrent() - _lastPlayheadTimeUpdated;
     float drift = playheadTimeElapsed - wallTimeElapsed;
+    
     // The playhead has to have moved > 500ms and we have to have signifigantly drifted in comparision to wall time.
     // We check both positive and negative to account for seeking forward and backward respectively.
     // Unbuffered seeks seem to update the playhead time when transitioning into play where as buffered seeks update the playhead time when paused.
     if (fabsf(playheadTimeElapsed) > MUXSDKMaxSecsSeekPlayheadShift &&
-        fabsf(drift) > MUXSDKMaxSecsSeekClockDrift &&
-        (_state == MUXSDKPlayerStatePaused || _state == MUXSDKPlayerStatePlay)) {
-        _seeking = YES;
-        MUXSDKInternalSeekingEvent *event = [[MUXSDKInternalSeekingEvent alloc] init];
-        MUXSDKPlayerData *playerData = [self getPlayerData];
-        if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomTV) {
-            [self setPlayerPlayheadTime:_lastPlayheadTimeMsOnPause onPlayerData:playerData];
+        fabsf(drift) > MUXSDKMaxSecsSeekClockDrift) {
+        if (_state == MUXSDKPlayerStatePaused || _state == MUXSDKPlayerStatePlay) {
+            _seeking = YES;
+            MUXSDKInternalSeekingEvent *event = [[MUXSDKInternalSeekingEvent alloc] init];
+            MUXSDKPlayerData *playerData = [self getPlayerData];
+            if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomTV) {
+                [self setPlayerPlayheadTime:_lastPlayheadTimeMsOnPause onPlayerData:playerData];
+            }
+            [event setPlayerData:playerData];
+            [MUXSDKCore dispatchEvent:event forPlayer:_name];
+        } else if (_state == MUXSDKPlayerStatePlaying) {
+            // Programatically called seek
+            _seeking = YES;
+            MUXSDKInternalSeekingEvent *seekingEvent = [[MUXSDKInternalSeekingEvent alloc] init];
+            MUXSDKPlayerData *playerData = [self getPlayerData];
+            [self setPlayerPlayheadTime:_lastPlayheadTimeMs onPlayerData:playerData];
```

Comment:
is there a reason that this doesn't match lines 893-895?

Reply:
Yes those lines of code were added because of the way the controller works on apple tv, when you're interacting with the apple tv to seek it pauses and then it was taking the incorrect playhead time. But the only way to get in this block is by calling seek programmatically so there is no special behavior on apple tv cause the seek just happens immediately that's why we send the seeking time as the last playhead time registered and then the seeked time is the current time.

Reply:
Do you actually need this line? I'm not sure the playhead time has to be explicitly be set here. It does in the case of apple tv because we are attempting to override the playhead with what it was on pause

Reply:
yes we need to set it because the seek happens immediately which means the seeking playhead would take the seeked playhead time from when the seeking ended instead of the start seeking one that's why we register the lastPlayheadTime and send that for the Seeking instead of leaving the current playhead time we get from playerData which is actually the seeked time.

Reply:
ooo I see